src/nti/site/site.py

```diff
@@ -314,51 +315,108 @@ def _btlar_after_setstate(self):
     # these ``_subscribers`` is to implement ``getAllUtilitiesRegisteredFor``; that's rarely called,
     # right? Maybe rare enough that we could implement a different mechanism that doesn't need to
     # persistently store the whole list at all.
+    #
+    # IMPORTANT: Once a parent mapping has become a BTree, all of its children mappings
+    # must ALSO become BTrees. This is because the mappings are mutated in-place as registrations
+    # are added or removed, and doing that doesn't let the BTree know that it needs to become
+    # ``_p_changed``; trying to figure out which child dictionary actually changed *could* be done
+    # (err, somehow) but we'd actually need to set the ``_p_changed`` on a BTree *bucket*,
+    # and those aren't directly accessible. So: once a BTree, it's BTrees all the way down.
+
+    def __mapping_needs_converted(self, path, mapping, parent_is_btree, btree_type):
+        # We need to convert if the parent is a btree (and we're not a btree already)
+        # OR we're not a btree and we're too long
+        regs_is_btree = isinstance(mapping, btree_type)
+
+        needs_converted = (
+            parent_is_btree and not regs_is_btree
+        ) or (
+            not regs_is_btree and len(mapping) > self.btree_map_threshold
+        )
```

Comment:
Worth simplifying this for readability?

```suggestion
        needs_converted=False
        if not regs_is_btree:
            needs_converted = parent_is_btree or len(mapping) > self.btree_map_threshold
```

Reply:
I find that less readable 😢

Reply:
I agree, I like the current form of the conditional.